layershellev/src/lib.rs

```diff
@@ -2429,33 +2415,41 @@ impl<T: 'static> WindowState<T> {
                                 exclusive_zone,
                                 margin,
                                 keyboard_interactivity,
-                                use_last_output,
                                 events_transparent,
+                                output_setting,
                             },
                             info,
                         )) => {
                             let pos = self.surface_pos();
 
-                            let mut output = pos.and_then(|p| self.units[p].wl_output.as_ref());
-
                             if self.last_wloutput.is_none()
                                 && self.outputs.len() > self.last_unit_index
                             {
                                 self.last_wloutput =
                                     Some(self.outputs[self.last_unit_index].1.clone());
                             }
 
-                            if use_last_output {
-                                output = self.last_wloutput.as_ref();
-                            }
+                            let output = match output_setting {
+                                events::LayerOutputSetting::None => {
+                                    pos.and_then(|p| self.units[p].wl_output.as_ref()).cloned()
+                                }
+                                events::LayerOutputSetting::FollowLastOutput => {
+                                    self.last_wloutput.clone()
+                                }
+                                events::LayerOutputSetting::ChosenOutput(output) => self
+                                    .xdg_info_cache
```

Comment:
`self.xdg_info_cache` in not always populated at this point
Following cycle must be either copied here or moved outside of if `is_allscreens` statement where it is already present:
```rust
for (_, output_display) in &self.outputs {
    let zxdgoutput = xdg_output_manager.get_xdg_output(output_display, &qh, ());
    self.xdg_info_cache
        .push((output_display.clone(), ZxdgOutputInfo::new(zxdgoutput)));
}
```

Reply:
> `self.xdg_info_cache` in not always populated at this point Following cycle must be either copied here or moved outside of if `is_allscreens` statement where it is already present:
> 
> ```rust
> for (_, output_display) in &self.outputs {
>     let zxdgoutput = xdg_output_manager.get_xdg_output(output_display, &qh, ());
>     self.xdg_info_cache
>         .push((output_display.clone(), ZxdgOutputInfo::new(zxdgoutput)));
> }
> ```
It should has be added at
https://github.com/waycrate/exwlshelleventloop/pull/86/files#diff-d794182b4533a00f543694b67814cc1ffca93afbc14c08c438cb67f8994c821eR2141-R2208